### PR TITLE
Add java source code formatting to gradle build

### PR DESCRIPTION
**Pull request by thombergs**
This PR should be merged at a time where the least possible amount of people are currently working on a Pull Request to be able to format all files and commit them.

This PR breaks the build! You have to call "gradlew spotlessApply" to format all files and then check them all in at once to have a clean code base.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,53 +1,39 @@
...
-    apply plugin: 'java'
-//    apply plugin: 'checkstyle'
-//    apply plugin: 'findbugs'
-//    apply plugin: 'jacoco'
-//    apply plugin: 'pmd'
+
+    buildscript {
+        repositories {
+            mavenCentral()
+            mavenLocal()
```
Comment (matthiasbalke) [inline build.gradle]: `mavenLocal()` should be listed before `mavenCentral()` as Gradle checks the repositories in definied order and stops at the first repository which contains the requested artifact.

File: build.gradle [hunk 2]
```diff
@@ -1,53 +1,39 @@
...
+            classpath "org.ajoberstar:gradle-git:1.5.1"
+            classpath "com.diffplug.spotless:spotless-plugin-gradle:3.0.0"
+            classpath "com.github.jacobono:gradle-jaxb-plugin:1.3.6"
+        }
+    }
+
+    apply plugin: "idea"
+    apply plugin: "java"
+
+    version "1.0.0-SNAPSHOT"
```
Comment (matthiasbalke) [inline build.gradle]: In a later PR you should switch to continuous versioning.

File: build.gradle [hunk 3]
```diff
@@ -1,53 +1,39 @@
...
-
-}
-// this task has been automatically added by coderadar to access all JAR dependencies for bytecode analysis
-task printClasspathForCoderadar {
-    doLast {
-        configurations.testRuntime.each { println it }
-    }
-}
-
-task wrapper(type: Wrapper) {
```
Comment (matthiasbalke) [inline build.gradle]: why did you remove the wrapper definition?

Comment (thombergs) [inline build.gradle]: because it was just needed to initialize the gradle wrapper and we don't need it anymore

File: build.gradle (hunk 2 above)
Comment (thombergs) [inline build.gradle]: added issue https://github.com/reflectoring/coderadar/issues/143

File: build.gradle (hunk 3 above)
Comment (matthiasbalke) [inline build.gradle]: Each time you call `gradlew wrapper` the wrapper is updated by the definition from the `build.gradle`. So you can define which wrapper version should be installed in your project without updating the `gradle-wrapper.properties` by hand.

Comment (thombergs) [inline build.gradle]: but i can just as well update the `gradle-wrapper.properties` by hand. I like the idea better to have a clean build.gradle.

File: build.gradle (hunk 1 above)
Comment (thombergs) [inline build.gradle]: changed the order

**Comment by thombergs** [comment]
please don't merge this PR! let me do that, once I have made sure that no one is currently working on a PR so that we can update formatting on the whole code base without conflicts
